### ISSUE TITLE
Freeze when using Audio Shield and DMX transmit on Serial3

### ISSUE THREAD

**Issue by ssilverman** [ssilverman/TeensyDMX#20]
Original discussion is here:
https://forum.pjrc.com/threads/71964-Using-Serial3-when-using-audio-library

In summary, it appeared that the Audio Shield was causing some freeze when transmitting DMX on Serial 3. Upon further digging, it was discovered that it was some interaction with TeensyDMX, plus the Audio Shield capacitor on pin 15 (RX3).

Test code:
```c++
#include <TeensyDMX.h>
// #include <Audio.h>

namespace teensydmx = ::qindesign::teensydmx;
teensydmx::Sender dmxTx{Serial3};

// AudioOutputI2S           audioOutput;

void setup() {
  Serial.begin(115200);
  delay(1500);
  if (CrashReport) {
    Serial.println(CrashReport);
    CrashReport.clear();
  } else {
    Serial.println("No CrashReport");
  }
  Serial.println("setup begin");

  dmxTx.begin();

  for (int i=0; i<25; i++) {
    Serial.println(i);
    delayMicroseconds(250);
  }
  Serial.println("setup end");
}

void loop() {
  static elapsedMillis timer;
  static uint32_t counter = 0;
  if (timer >= 1000) {
    Serial.println(counter++);
    timer = 0;
  }
}
```

The cause is a capacitor connecting the RX line to voltage, eventually triggering RX and causing continuous TX. i.e. When RX reaches a threshold. The best scope trace that shows this is in post # 15:
https://forum.pjrc.com/threads/71964-Using-Serial3-when-using-audio-library?p=319218&viewfull=1#post319218
![file](https://user-images.githubusercontent.com/2762930/213951092-de6ac6c6-96ec-4c83-b718-0a421e9d6af8.png)

The blue trace shows that the TX interrupt seems to be continuously triggering after RX reaches some threshold. Code to generate this (on `Serial1`):
```c++
void lpuart6_tx_isr() {
digitalWriteFast(32, HIGH); // Blue trace on scope
  Sender *s = txInstances[0];
  if (s != nullptr) {
    s->sendHandler_->irqHandler();
  }
digitalWriteFast(32, LOW);
}
```

(Credit to @PaulStoffregen.)

**Comment by ssilverman** [comment]
After some digging and looking at the STAT register to see which bits were set, the following details came to light:
1. Setting the active/inactive/completing states in `LPUARTSendHandler` only changed necessary flags and left all others unchanged, unlike `UARTSendHandler` which sets all the bits in the C2 control register, and
2. The BREAK/slot serial parameters were saved immediately after setting up the serial port, meaning the RX options, including interrupt-enables, were set too.

This means that the RX options were getting reset whenever the BREAK and slot serial parameters were getting restored.

The solution was to disable all the RX options before saving the serial parameters.

**Comment by ssilverman** [comment]
Fixed in 69926c20823763af9c482f8bd578701e60563ad4.

**Comment by ssilverman** [comment]
Fixed in 69926c20823763af9c482f8bd578701e60563ad4.
